### ISSUE TITLE
nolint for function and for line

### ISSUE THREAD

**Issue by rekby** [golangci/golangci-lint#106]
for code:

```golang
package main

import (
	"bytes"
	"io"
)

//nolint:gas
func main() {
	var buf io.Writer = &bytes.Buffer{}
	//nolint:errcheck
	buf.Write([]byte("123"))
}
```

I have error errcheck
```
golangci-lint run
main.go:12:11: Error return value of `buf.Write` is not checked (errcheck)
        buf.Write([]byte("123"))
```

Expected no errors: 
```
Congrats! No issues were found.
```

**Comment by jirfag** [comment]
@rekby thank you! cool bug :)
@alecthomas FYI gometalinter has the same bug:
```
~/go/src/github.com/golangci/golangci-lint (master *)$ cat pkg/result/processors/testdata/nolint2.go
package testdata

import (
        "bytes"
        "io"
)

func noTransitiveExpanding() {
        //nolint:errcheck
        var buf io.Writer = &bytes.Buffer{}
        buf.Write([]byte("123"))
}

func nolintFuncByInlineCommentDoesNotWork() { //nolint:errcheck
        var buf io.Writer = &bytes.Buffer{}
        buf.Write([]byte("123"))
}
~/go/src/github.com/golangci/golangci-lint (master *)$ vim test.go
~/go/src/github.com/golangci/golangci-lint (master *)$ gometalinter --disable-all -E errcheck test.go
~/go/src/github.com/golangci/golangci-lint (master *)$  golangci-lint run --disable-all -E errcheck test.go
ERRO Running error: --enable-all and --disable-all options must not be combined
~/go/src/github.com/golangci/golangci-lint (master *)$  golangci-lint run --disable-all -E errcheck --no-config test.go
test.go:11:12: Error return value of `buf.Write` is not checked (errcheck)
  buf.Write([]byte("123"))
           ^
~/go/src/github.com/golangci/golangci-lint (master *)$ cat test.go
package testdata

import (
  "bytes"
  "io"
)

func noTransitiveExpanding() {
  //nolint:errcheck
  var buf io.Writer = &bytes.Buffer{}
  buf.Write([]byte("123"))
}
~/go/src/github.com/golangci/golangci-lint (support/fix-nolint-transitive-expanding)$ gometalinter --disable-all -E errcheck test.go
~/go/src/github.com/golangci/golangci-lint (support/fix-nolint-transitive-expanding)$ golangci-lint run --disable-all -E errcheck --no-config test.go
test.go:11:12: Error return value of `buf.Write` is not checked (errcheck)
  buf.Write([]byte("123"))
           ^
```

**Comment by septs** [comment]
edge cases:
```go
var buf bytes.Buffer
binary.Write(&buf, binary.BigEndian, ....)
v.WriteTo(&buf)
// etc
```
